### PR TITLE
Added docs and local helm registry

### PR DESCRIPTION
**Pull request by Algeran**
- Simple docker registry is deployed to the kind cluster as OCI Helm registry, exposed to the local machine with the NodePort on 30500
- Added docker daemon installation on Linux-based systems
- Added notes about `make help` to the README.md
- Added docs with detailed explanation for General setup steps
- Fixed helm package bug with missing templates directory (added .gitkeep files)

### CODE REVIEW

File: setup/helmRepository.yaml [hunk 1]
```diff
@@ -1,3 +1,37 @@
+---
+apiVersion: v1
+kind: Pod
```
Comment (soerenschneider) [inline setup/helmRepository.yaml:3]: Nit: Using a `Pod` directly will work with the given restartPolicy, but why not just use a Deployment?

File: setup/helmRepository.yaml [hunk 2]
```diff
@@ -1,3 +1,37 @@
...
+  name: helm-registry
+  namespace: hmc-system
+spec:
+  containers:
+  - image: registry:2
+    name: helm-registry
+    ports:
+    - containerPort: 5000
+  dnsPolicy: ClusterFirst
+  restartPolicy: Always
```
Comment (soerenschneider) [inline setup/helmRepository.yaml:16]: Just a thought, should we add (simple) persistence by using a (e.g. hostpath) volumes to store charts across reboots?

File: setup/helmRepository.yaml [hunk 3]
```diff
@@ -10,4 +44,4 @@ spec:
   interval: 10m0s
   provider: generic
   type: oci
-  url: oci://ghcr.io/mirantis/2a-demos/charts
+  url: oci://helm-registry:5000/helm-charts
```
Comment (soerenschneider) [inline setup/helmRepository.yaml:47]: I'm not sure if we should rather use a variable with a default of `oci://ghcr.io/mirantis/2a-demos/charts` here as this should only be used for local development, right? Or is the idea to abandon all charts currently stored in the GitHub repo as well?

File: Makefile [hunk 4]
```diff
@@ -469,38 +487,16 @@ package-chart-%: lint-chart-%
...
-					echo "REGISTRY_USERNAME and REGISTRY_PASSWORD must be populated to push the chart to an HTTPS repository"; \
-					exit 1; \
-				else \
-					$(HELM) repo add hmc $(REGISTRY_REPO); \
-					echo "Pushing $$chart to $(REGISTRY_REPO)"; \
-					$(HELM) cm-push "$$chart" $(REGISTRY_REPO) --username $$REGISTRY_USERNAME --password $$REGISTRY_PASSWORD; \
-				fi; \
-			fi; \
+			echo "Pushing $$chart to $(REGISTRY_REPO)"; \
+			$(HELM) push "$$chart" $(REGISTRY_REPO); \
```
Comment (soerenschneider) [inline Makefile:500]: This is going to break the [CI pipeline job](https://github.com/Mirantis/2a-demos/blob/6e86365e978ae9bb0e04706e66dbb5cbd566089a/.github/workflows/release.yml#L26-L27) that actually pushes the charts to the GitHub container registry.

This comment kind of related to my other comment in the helmRepository.yaml file, I don't know what's the plan forward with the charts in the repo. Maybe @Schnitzel has a strong(er) opinion here.

**Review by soerenschneider — COMMENTED** [review]
LGTM in general, however question(s) about the helmRepo remain.

File: setup/helmRepository.yaml (hunk 2 above)
Comment (Algeran) [reply setup/helmRepository.yaml:16]: Good point. I added the PVC that automatically creates the volume with the default kind local-path provisioner.

File: setup/helmRepository.yaml (hunk 1 above)
Comment (Algeran) [reply setup/helmRepository.yaml:3]: It's just an auto-generated yamls with the `kubectl run --restart Always --expose ...` command and I was a bit lazy to replace it with the Deployment 😄 Replaced with the Deployment

File: setup/helmRepository.yaml (hunk 3 above)
Comment (Algeran) [reply setup/helmRepository.yaml:47]: >Or is the idea to abandon all charts currently stored in the GitHub repo as well?

Yes, it should use only local registry with charts that are generated and pushed with the `make push-helm-charts` command (I added it to the README.md as well). Using Github repo as OCI registry is strictly bound us to the particular repo and can add various problems in the future if someone decides to fork this repo or create other charts/change existing

File: Makefile (hunk 4 above)
Comment (Algeran) [reply Makefile:500]: It actually doesn't matter, we should not use Github repo as OCI registry. I forgot to remove github actions that push charts, fixed this

**Comment by soerenschneider** [comment]
LGTM, you can merge it.

**Pull request merged**